src/dbally/data_models/prompts/query_explainer_prompt_template.py

```diff
@@ -15,37 +21,30 @@ def __init__(
         response_format: Optional[Dict[str, str]] = None,
         llm_response_parser: Callable = lambda x: x,
     ) -> None:
-        """
-        Initializes IQLExplainerPromptTemplate class.
-
-        Args:
-            chat: chat format
-            response_format: response format
-            llm_response_parser: function to parse llm response
-        """
-
         super().__init__(chat, response_format, llm_response_parser)
-        self.chat = _check_prompt_variables(chat, {"question", "filters"})
+        self.chat = _check_prompt_variables(chat, {"question", "query", "number_of_results"})
 
 
-default_iql_explainer_template = IQLExplainerPromptTemplate(
+default_query_explainer_template = QueryExplainerPromptTemplate(
     chat=(
         {
             "role": "system",
             "content": "You are a helpful assistant that helps describe a table generated by a query composed "
-            "of a list of filters that answers users' question. "
+            "that answers users' question. "
             "You are very intelligent and obedient.\n"
             "Your task is to provide natural language description of the table used by the logical query "
             "to the database.\n"
             "Describe the table in a way that is short and informative.\n"
-            "Make your answer as short as possible and start it by infroming the user that the table is too long "
-            "to be used in LLM call due to token limit.\n"
+            "Make your answer as short as possible and start it by infroming the user that the data is too long "
+            "to be used in LLM call due to token limit and then describe the table "
```

Comment:
In my tests the new part ("and then describe the table based on the question and the query") is required for the model to produce anything beyond "The data is too long to be used in LLM call".

Now the model produces a simplistic description of the table ("The table contains 23 employees from France").

Reply:
Ok, I'll try this version.

More generally I think we should make about improving this scenario from a Data Science view point. When the model is given the data the answers are really good, but the fallback answers are extremely primitive (you could probably just make them with string substitution, without using LLM at all). For now, I wanted to avoid opening the Pandora's box, so I tried to change the prompt as little as humanly possible ;)

Reply:
I tried your version, seems to work reasonably okay so I left it in.